src/TuDa.CIMS.Web/Components/Pages/ShoppingCartPage.razor.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNetCore.Components;
+using MudBlazor;
+using TuDa.CIMS.Shared.Entities;
+using TuDa.CIMS.Web.Components.ShoppingCart;
+
+namespace TuDa.CIMS.Web.Components.Pages;
+
+public partial class ShoppingCartPage
+{
+    [Inject]
+    public required IDialogService DialogService { get; set; }
+
+    private Purchase Purchase { get; set; } =
+        new()
+        {
+            Buyer = new Student { FirstName = "John", Name = "John Doe" },
+        };
+
+    public required AssetItem Product { get; set; }
+
+    private async Task OpenDialogAsync(AssetItem assetItem)
+    {
+        Product = assetItem;
+
+        var options = new DialogOptions { CloseOnEscapeKey = true };
+
+        // Set Parameters
+        var parameters = new DialogParameters { { "Product", assetItem } };
+
+        var dialog = await DialogService.ShowAsync<ShoppingCartProductDialog>(
+            "Amount Input",
+            parameters,
+            options
+        );
+        var result = await dialog.Result;
+
+        if (result is { Canceled: false })
+        {
+            int? amount = result.Data as int?;
+            if (amount > 0)
+            {
+                AddProductEntry(amount.Value);
+            }
+        }
+    }
+
+    private async Task OpenSubmitDialogAsync()
+    {
+        var options = new DialogOptions { CloseOnEscapeKey = true };
+        var parameters = new DialogParameters
+        {
+            { "PurchaseEntries", Purchase.Entries },
+            {
+                "WorkingGroups",
+                new List<WorkingGroup>()
+                {
+                    new() { Professor = new Professor { Name = "Heiter" } },
+                    new() { Professor = new Professor { Name = "Kaiser" } },
+                }
+            },
+        };
+
+        var dialog = await DialogService.ShowAsync<ShoppingCartSubmitPopup>(
+            "Submit Input",
+            parameters,
+            options
+        );
+
+        if (await dialog.GetReturnValueAsync<WorkingGroup>() is not null)
+        {
+            Purchase = new Purchase { Buyer = new Student { Name = "Jon" } };
+        }
```

Comment:
🔧 You need to use the resulted `WorkingGroup` here and set the `WorkingGroup` of the Purchase here and not creating a new instance.
```suggestion
       var workingGroup = await dialog.GetReturnValueAsync<WorkingGroup>();
        if (workingGroup is not null)
        {
            Purchase.WorkingGroup = workingGroup;
        }
```

Reply:
Purchase does not have a working group anymore only a Buyer. Changes are needed in Purchase or in ShoppingCartSubmitPopup.

Reply:
You're right. We will need to extract the sending of the purchase to the API to another ticket, because we need the ticket #129